src/proc/y16i-to-y10msby10msb.cpp

```diff
@@ -11,10 +11,15 @@
 namespace librealsense
 {
     struct y16i_pixel { uint16_t left : 16, right : 16;
-                        // left and right data are shifted 6 times so that the MSB 10 bit will hold the data
-                        // data should be in MSB because OpenGL then uses the MSB byte
-                        uint16_t l() const { return left << 6; }
-                        uint16_t r() const { return right << 6; } };
+                        // explanation of "return x << 6 | x >> 4" :
+                        // Since the data is received only in 10 bits, and the conversion is to 16 bits, 
+                        // the range moves from [0 : 2^10-1] to [0 : 2^16-1], so the values should be converted accordingly:
+                        // x is range [0 : 2^10-1] is converted to y = x * (2^16-1)/(2^10-1) approx= x * (64 + 1/16)
+                        // And x * (64 + 1/16) = x * 64 + x * 1/16 = x << 6 | x >> 4
+                        // This operation is done using shiftings to make it more efficient.
```

Comment:
Nice!
I would add 
`This operation is done using shiftings to make it more efficient, with a non significant accuracy loss.`
Sounds OK?

Reply:
yep